schema/schemaspec/schemahcl/hcl.go

```diff
@@ -70,12 +69,69 @@ func (s *State) UnmarshalSpec(data []byte, v interface{}) error {
 	return s.Eval(data, v, nil)
 }
 
-// Eval implements the Evaluator interface.
-func (s *State) Eval(data []byte, v interface{}, input map[string]string) error {
-	spec, err := s.eval(data, input)
+// EvalFiles evaluates the files in the provided paths using the input variables and
+// populates v with the result.
+func (s *State) EvalFiles(paths []string, v interface{}, input map[string]string) error {
+	parser := hclparse.NewParser()
+	for _, path := range paths {
+		_, diag := parser.ParseHCLFile(path)
+		if diag.HasErrors() {
+			return diag
+		}
+	}
+	return s.eval(parser, v, input)
+}
+
+// eval evaluates the parsed HCL documents using the input variables and populates v
+// using the result.
+func (s *State) eval(parsed *hclparse.Parser, v interface{}, input map[string]string) error {
+	ctx := s.config.newCtx()
+	reg := &blockDef{
+		fields:   make(map[string]struct{}),
+		children: make(map[string]*blockDef),
+	}
+	files := parsed.Files()
+	fileNames := make([]string, 0, len(files))
+	allBlocks := make([]*hclsyntax.Block, 0)
```

Comment:
I'd assume there's a least one block in each file. 
```suggestion
	allBlocks := make([]*hclsyntax.Block, 0, len(files))
```

But in case you want to keep it zero, use nil array instead

```go
var allBlocks []*hclsyntax.Block
```